### PR TITLE
Add Gatsby/Builder example

### PR DESCRIPTION
**Pull request by teleaziz**

### CODE REVIEW

File: examples/gatsby-builder-example-no-plugin/LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+The MIT License (MIT)
+
+Copyright (c) 2018 Resi Respati
```
Comment (steve8708) [inline examples/gatsby-builder-example-no-plugin/LICENSE]: `Copyright (c) 2018 Builder.io, Inc`

File: examples/gatsby-builder-example-no-plugin/README.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
+<p align="center">
```
Comment (steve8708) [inline examples/gatsby-builder-example-no-plugin/README.md]: needs a new readme (I know you have it planned, just adding comment here to track it)

File: examples/gatsby-builder-example-no-plugin/gatsby-config.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  siteMetadata: {
+    title: 'gatsby-starter-typescript-plus',
+    description:
+      'A starter kit for TypeScript-based Gatsby projects with sensible defaults.',
+    keywords: 'gatsbyjs, gatsby, javascript, sample, something',
+    siteUrl: 'https://gatsby-starter-typescript-plus.netlify.com',
+    author: {
+      name: 'Aziz Abbas',
+      url: 'https://twitter.com/resir014',
+      email: 'aziz@builder.io'
```
Comment (steve8708) [inline examples/gatsby-builder-example-no-plugin/gatsby-config.js]: these fields could use some updating (e.g. "builder gatsby example") etc

File: examples/gatsby-builder-example-no-plugin/gatsby-config.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    }
+  },
+  plugins: [
+    {
+      resolve: 'gatsby-plugin-canonical-urls',
+      options: {
+        siteUrl: 'https://www.builder.io/docs'
```
Comment (steve8708) [inline examples/gatsby-builder-example-no-plugin/gatsby-config.js]: may want to just remove this plugin for now

File: examples/gatsby-builder-plugin-example/plugins/gatsby-plugin-builder/src/gatsby-node.js [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+
+exports.createPages = async ({ graphql, actions }, options) => {
+  const config = {
+    ...defaultOptions,
+    ...options
+  };
+  const { createPage } = actions;
+  const models = Object.keys(config.templates);
+  const result = await graphql(`
+    query allBuilders {
```
Comment (steve8708) [inline examples/gatsby-builder-plugin-example/plugins/gatsby-plugin-builder/src/gatsby-node.js]: `allBuilders` could maybe use a different name (e.g. `allBuilderPages`?) or I think you can leave the name off in gql queries of this type - aka just `query { ... }`

File: examples/gatsby-builder-plugin-example/plugins/gatsby-plugin-builder/src/gatsby-node.js [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+  models.forEach(modelName => {
+    result.data[config.fieldName][modelName].forEach(entry => {
+      if (
+        entry.everything.data.url &&
+        entry.everything.published === 'published'
+      ) {
+        const component = config.templates[modelName];
+        invariant(
+          component && component.length > 0,
+          `gatsby-plugin-builder requires a template path for each model`
```
Comment (steve8708) [inline examples/gatsby-builder-plugin-example/plugins/gatsby-plugin-builder/src/gatsby-node.js]: this seems plugin related?

File: examples/gatsby-builder-plugin-example/plugins/gatsby-plugin-builder/src/transforms.js [hunk 7]
```diff
@@ -0,0 +1,62 @@
+const { GraphQLObjectType, GraphQLNonNull } = require(`gatsby/graphql`);
```
Comment (steve8708) [inline examples/gatsby-builder-plugin-example/plugins/gatsby-plugin-builder/src/transforms.js]: can we just use the graphql plugin in this example and not have to include all of this stuff?

File: examples/gatsby-builder-example-no-plugin/src/layouts/index.tsx [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+// interface StaticQueryProps {
+//   site: {
+//     siteMetadata: {
+//       title: string;
+//       description: string;
+//       keywords: string;
+//     };
+//   };
+// }
```
Comment (steve8708) [inline examples/gatsby-builder-example-no-plugin/src/layouts/index.tsx]: remove?

File: examples/gatsby-builder-example-no-plugin/.eslintrc [hunk 9]
```diff
@@ -0,0 +1,27 @@
+{
```
Comment (steve8708) [inline examples/gatsby-builder-example-no-plugin/.eslintrc]: let's call the folder this lives in just `gatsby`

later we can update this as the canonical gatsby example the plugin when the plugin is good to go

**Comment by steve8708** [comment]
realized I didn't catch that this has both plugin and no plugin examples

let's implify this to just be `exmples/gatsby` and for now be the absolute simplest example we can provide just using graphql (e.g. with the graphql plugin) and no builder plugin

separately we can do the PR for the plugin and update this as the canonical example of using gatsby with builder and utilize the plugin

**Comment by teleaziz** [comment]
Removed the plugin example and addressed feedback

**Pull request merged**